dowhy/causal_model.py

```diff
@@ -308,26 +308,17 @@ def estimate_effect(
                 causal_estimator = self._estimator_cache[method_name]
             else:
                 causal_estimator = causal_estimator_class(
-                    self._data,
                     identified_estimand,
-                    self._treatment,
-                    self._outcome,  # names of treatment and outcome
-                    control_value=control_value,
-                    treatment_value=treatment_value,
-                    test_significance=test_significance,
-                    evaluate_effect_strength=evaluate_effect_strength,
-                    confidence_intervals=confidence_intervals,
-                    target_units=target_units,
-                    effect_modifiers=effect_modifiers,
                     **method_params,
                     **extra_args,
                 )
+
                 self._estimator_cache[method_name] = causal_estimator
 
         return estimate_effect(
+            self._data,
             self._treatment,
             self._outcome,
-            identified_estimand,
             identifier_name,
             causal_estimator,
             control_value,
```

Comment:
extra_args should also be passed to estimate_effect? Right now, they are ignored.

Reply:
`extra_args` is only used for `init_params` which is used to instantiate estimators, method_params are the extra parameters for executing the estimate_effect method